l10n_nl_tax_invoice_basis/models/account_tax.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2017-2018 Onestein (<https://www.onestein.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+from odoo.osv import expression
+
+
+class AccountTax(models.Model):
+    _inherit = 'account.tax'
+
+    def get_move_line_partial_domain(self, from_date, to_date, company_id):
+        res = super().get_move_line_partial_domain(
+            from_date,
+            to_date,
+            company_id
+        )
+
+        if self.env.context.get('skip_invoice_basis_domain'):
+            return res
+
+        company = self.env['res.company'].browse(company_id)
+        if company.country_id != self.env.ref('base.nl'):
+            return res
+        if not company.l10n_nl_tax_invoice_basis:
+            return res
+
+        domain_params = {
+            'company_id': company_id,
+            'from_date': from_date,
+            'to_date': to_date,
+        }
+        # following line breaks the inheritance chain;
+        # it is intentional, to avoid other modules to interfere;
+        # pass context ''skip_invoice_basis_domain' if you
+        # don't want to allow this behavior
+        return self._get_invoice_basis_domain(domain_params)
+
+    def _get_invoice_basis_domain(self, domain_params):
+        domain_company = [('company_id', '=', domain_params['company_id'])]
```

Comment:
According to https://github.com/OCA/account-financial-reporting/pull/619 (that contains https://github.com/OCA/account-financial-reporting/pull/613), here you should pass `company_ids`.

```python
domain_company = [('company_id', 'in', domain_params['company_ids'])]
```

And the previous method should be written this way:

```python
    def get_move_line_partial_domain(self, from_date, to_date, company_ids):
        res = super().get_move_line_partial_domain(
            from_date,
            to_date,
            company_ids
        )

        if self.env.context.get('skip_invoice_basis_domain'):
            return res

        companies = self.env['res.company'].browse(company_ids)
        countries = companies.country_id
        if any(country != self.env.ref('base.nl') for country in countries):
            return res

        domain_params = {
            'company_ids': company_ids,
            'from_date': from_date,
            'to_date': to_date,
        }
        # following line breaks the inheritance chain;
        # it is intentional, to avoid other modules to interfere;
        # pass context ''skip_invoice_basis_domain' if you
        # don't want to allow this behavior
        return self._get_invoice_basis_domain(domain_params)

```